src/hotspot/os/windows/os_windows.hpp

```diff
@@ -56,6 +63,38 @@ class os::win32 {
   // Windows-specific interface:
   static void   initialize_system_info();
   static void   setmode_streams();
+  static bool   is_windows_11_or_greater();
+  static bool   is_windows_server_2022_or_greater();
+  static DWORD  system_logical_processor_count();
+  static int windows_major_version() {
+    initialize_windows_version();
+    return _major_version;
+  }
+  static int windows_minor_version() {
+    initialize_windows_version();
+    return _minor_version;
+  }
+  static int windows_build_number() {
+    initialize_windows_version();
+    return _build_number;
+  }
+  static int windows_build_minor() {
+    initialize_windows_version();
+    return _build_minor;
+  }
```

Comment:
There should only be one path that initializes the version info and we should be very clear what that one path is. Can these methods be called before the call to `os::init` that calls `initialize_system_info` that calls `initialize_windows_version`? If so we should fix that. Otherwise these methods should assert the version is initialized.

Then `initialize_windows_version` should assert that it is not already initialized.